scripts/document_updater.py

```diff
@@ -0,0 +1,97 @@
+import sys
+import re
+
+from typing import List
+from dataclasses import dataclass
+from pathlib import Path
+from inspect import cleandoc
+
+from exasol_advanced_analytics_framework.example import generator as example_generator
+
+
+class ParseException(Exception):
+    """
+    If input file is not well-formed.
+    """
+
+
+@dataclass
+class Template:
+    path: str
+
+    def render(self):
+        if self.path != "example/generator.py":
+            raise ParseException("document_updater.Template currently only"
+                                 " supports path example/generator.py")
+        return "\n".join([
+            "<!-- The example is deliberately tagged as language python since the major",
+            "     parts are in python. Formally, however, the python code is embedded into",
+            "     an SQL statement, though. -->",
+            "```python",
+            example_generator.create_script(),
+            "",
+            example_generator.execute_script() + ";",
+            "```",
+            ""])
+
+
+class ChunkReader:
+    """
+    Enables to replace chunks of a string by generated text, e.g. from
+    jinja templates.
+    """
+    def __init__(self):
+        self._generated = None
+        self._plain = []
+        self._chunks = []
+
+    def _process_plain(self):
+        if self._plain:
+            self._chunks.append("\n".join(self._plain) + "\n")
+        self._plain = []
+
+    def _start_generated(self, line: str, match: re.Match):
+        if self._generated:
+            raise ParseException(
+                f"Found another {line} before {self._generated} was closed."
+            )
+        self._plain += [
+            line,
+            "<!-- Do not edit the text from here until /generated! -->",
+        ]
+        self._process_plain()
+        self._generated = line
+        self._chunks.append(Template(match.group(2)))
+
+    def _end_generated(self, line: str):
+        if not self._generated:
+            raise ParseException(
+                f"Found {line} before any <!-- generated from/by ... -->."
+            )
+        self._generated = None
+        self._plain.append(line)
+
+    def split(self, content: str) -> List[str|Template]:
+        start = re.compile("<!-- +generated +(from|by) +([^ ]+) +-->")
+        end = re.compile("<!-- +/generated +-->")
+        for line in content.splitlines():
+            match = start.match(line)
+            if match:
+                self._start_generated(line, match)
+            elif end.match(line):
+                self._end_generated(line)
+            elif not self._generated:
+                self._plain.append(line)
+        self._process_plain()
+        return self._chunks
+
+    @classmethod
+    def chunks(cls, content: str):
```

Comment:
Question:
Can I use this class like this
```
my_chunk_reader = ChunkReader()
my_split_content1 = my_chunk_reader.split(my_content1)
my_split_content2 = my_chunk_reader.split(my_content2)
...
```
Probably not, right? But you need to look at the code to realize this is not a good idea.

Reply:
The chunk-reader is only used for updating the examples in the user guide. 
I think you are right, that the sample usage sketched by you would create unexpected results. 
I see 2 options: 
* O1) change the implementation to make such a use impossible, e.g. by using static methods, splitting in the constructor, etc. 
* O2) Enable the reuse of a `ChunkReader` instance as indicated by your example. 

O2 seemed easier to me, so please see next push for an improved version.

Reply:
I must also admit that there currently not tests for `ChunkReader`, yet.

I ommitted the tests for the following reasons:
* `ChunkReader` is only used for updating the examples in the user guide
* `ChunkReader` is not mission-critical in my eyes, as you could also update the examples manually as a fall-back
* `ChunkReader` in my eyes could be rated as a kind of build-script similar to the GitHub workflows

For the GitHub workflows there are not tests, either.  On the other hand GitHub workflows as well as the document updater are used in each CI build and hence we have evidence that they are working and potential problems will be detected early.

But if all this doesn't convince you, then maybe there is an even better way to achieve the goals listed in the ticket #190? In this case, I am happy to discuss.

Reply:
My solutions to the problem with the design of this class would be one of those:
1. Make the `split` function clean up after itself, so it would be safe to call it multiple times.
2. Make this class private, i.e. `_ChunkReader` and make a function that uses this class like the `update_example`, or maybe the `update_example` itself is all you need.
3. Put it in the class docstring that this class should not be used like in my comment and should be used like in the `update_example`.

Reply:
Thanks to @tkilias for inspiring discussion yesterday.

I changed the implementation to 
* go without `ChunkReader`
* without updating the user guide
* replacing the scripts in the user guide by simple references to separate files 
* using these files in the integration test
* only replacing the names of the DB schema and BucketFS connection with `string.replace()`

I think this approach is much simpler, requires less code, no check if the examples in the user guide are out-dated.

Please see latest push.